File: solutions/ruby/0001_two-sum.rb
# @param {Integer[]} nums
# @param {Integer} target
# @return {Integer[]}
def two_sum(nums, target)
  hash = {}
  nums.each_with_index do |n, i|
    her = target - n
    return [hash[her], i] unless hash[her].nil?
    hash[n] = i
  end
end

File: solutions/golang/0001_two-sum.go
func twoSum(nums []int, target int) []int {
	m := make(map[int]int)
	for i := 0; i < len(nums); i++ {
		her := target - nums[i]
		if herIndex, ok := m[her]; ok {
			return []int{herIndex, i}
		}
		m[nums[i]] = i
	}
	return nil
}
